### PR TITLE
Fixed installed vpn services are not updated for browser update

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/36210

If there is installed services, it should be updated during the browser update
to make these service has latest properties such as their executable path.

Resolves

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

See the linked issue

Another test plan:
1. Uninstall brave
2. Install brave and check vpn services(`BraveNightlyVpnWireguardService` and `BraveNightlyVpnService`) are not registered
   Can check installed service from `Task Manager` -> `Services`.

### CODE REVIEW

**Review by goodov — APPROVED** [review]
`chromium_src`, `DEPS` lgtm.

File: browser/brave_vpn/win/service_details.cc [hunk 1]
```diff
@@ -144,17 +144,9 @@ std::wstring GetBraveVpnWireguardTunnelServiceName() {
   NOTREACHED_NORETURN();
 }
 
-base::FilePath GetBraveVPNWireguardServiceInstallationPath(
```
Comment (bsclifton) [inline browser/brave_vpn/win/service_details.cc:147]: Nice to remove this - somewhere along the way, this wasn't called anymore and we had missed it

Comment (bsclifton) [reply browser/brave_vpn/win/service_details.cc:147]: Looks like it was removed in original PR https://github.com/brave/brave-core/pull/20754/files 😄👍

**Review by bsclifton — APPROVED** [review]
Perfect! Helped talk out the logic with @simonhong when problem was ID'ed. Just finished reviewing and with this latest approach (which fixes pathing; as pathing can be different when inside mini-installer; ex: we need to use `target_path`) we're in great shape now. @simonhong confirmed the scenarios he tested with the CI build. Nice work! 😄

**Comment by MadhaviSeelam** [comment]
Verification `PASSED` using
```
Brave | 1.65.42 Chromium: 122.0.6261.94 (Official Build) nightly (64-bit)
-- | --
Revision | 169e4fc2ee1896cd009c59e7b537220ba880a882
OS | Windows 11 Version 23H2 (Build 22631.3155)
```

1. installed 1.65.6
2. launched Brave
3. opened brave://settings/system and verified Use Wireguard protocol in Brave VPN toggleOFF
4. visited account.bravesoftware.com in a new tab
5. entered basic-auth credentials
6. entered vpnpr21934feb28@mailinator.com, clicked Get login link
7. clicked the link in the email
8. verified credentials are loaded
9. connected to VPN via VPN panel 
10. confirmed `Use WireGuard protocol in Brave VPN` toggle is `On` 
11. confirmed taskbar wireguard tray icon shows connected status
12. confirmed `BraveVpnNightlyWireguardTunnelService` is running taskmanager
13. upgraded to 1.65.42 via `brave://settings/help`
14. confirmed Wireguard VPN is still connected
15. opened settings->Apps->Startup
16. clicked brave_vpn_wireguard_service.exe tab ->open properties from context menu ->Location
17. confirmed registered wireguard service's property in the path shows correct Brave version`1.64.42`
18. opened Registry Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Run
19. confirmed registry value shows with correct Brave version  
20. uninstalled Brave nightly
21. installed 1.65.42 
22. opened Task Manager -> Services
23. confirmed vpn services (`BraveNightlyVpnWireguardService` and `BraveNightlyVpnService`) are not registered

step 3| step 8 | step 9 | step 11| step 13a| step 13b| step 14| step 15-17| step 18-19
----- | ----- | ----- | ----- | ----- | ------- | ------ | ------ | ----
image|image|image|image|image|image|image|image|image|

***_Un install_***

step 21 | step 23
---- | -----
image|image

**Pull request merged**